src/main/cpp/src/api/annotation_service.cc

```diff
@@ -0,0 +1,274 @@
+                                                     const char *value,
+                                                     const int32_t value_length,
+                                                     const int bcf_ht_type) {
+  std::string genomic_field_name = data_source + "_" + info_attribute;
+  size_t bytes = bcf_ht_type_to_nbytes.at(bcf_ht_type)*value_length;
+  if (m_annotation_buffer_remaining >=  bytes) {
+    size_t start = m_annotation_buffer_size - m_annotation_buffer_remaining;
+    memcpy(&m_annotation_buffer[start], value, bytes);
+    m_annotation_buffer_remaining -= bytes;
+    return genomic_field_t(genomic_field_name, &m_annotation_buffer[start], value_length);
+  } else {
+    throw GenomicsDBException(logger.format("Annotation Buffer size={} specified to store genomic field info is too small", m_annotation_buffer_size));
+  }
+}
+
+/**
+  Annotate a genomic location using all of the configured data sources. Annotations
+  are stored in the genomic_fields vector with the name of the field set to the result of
+  concatonating the dataSource (ie ClinVar) with a separator (see AnnotationService.DATA_SOURCE_FIELD_SEPARATOR),
+  and the INFO field label.
+ */
+void AnnotationService::annotate(genomic_interval_t& genomic_interval, std::string& ref, const std::string& alt, std::vector<genomic_field_t>& genomic_fields) {
+  assert(m_annotation_buffer_size); // Should have been initialized in constructor
+  m_annotation_buffer_remaining = m_annotation_buffer_size;
+
+  for(auto annotation_source: m_annotation_sources) {
+    htsFile *htsfile_ptr = hts_open(annotation_source.filename.c_str(), "r");
+    VERIFY2(htsfile_ptr!=NULL, logger.format("Could not hts_open {} file in read mode", annotation_source.filename));
+
+    // Only supporting vcf files now
+    enum htsExactFormat format = hts_get_format(htsfile_ptr)->format;
+    VERIFY2(format==vcf, logger.format("File {} is not VCF (a)", annotation_source.filename));
+
+    bcf_hdr_t *hdr = bcf_hdr_read(htsfile_ptr);
+    VERIFY2(hdr!=NULL, logger.format("Could not read header in file {}", annotation_source.filename));
+
+    // Read the tbi index file to help with reading the annotation vcfs
+    tbx_t *tbx = tbx_index_load3(annotation_source.filename.c_str(), NULL, 0);
+    VERIFY2(tbx, logger.format("Could not load the index file for {}", annotation_source.filename));
+
+    // Query using chromosome and position range
+    std::string query_range = genomic_interval.contig_name + ":" + std::to_string(genomic_interval.interval.first) + "-" + std::to_string(genomic_interval.interval.second);
+    hts_itr_t *itr = tbx_itr_querys(tbx, query_range.c_str());
+
+    VERIFY2(itr, "Could not obtain tbx query iterator. Possibly caused by the vcf not having any variants on the requested chromosome.");
+
+    // I'm not sure what this does. Whatever it is, it takes a really long time.
+    // Need to look in to how to remove this variable.
```

Comment:
Looks like the comment starting "I'm not sure" has been addressed so we can remove it and the commented-out code.